### PR TITLE
SECURITY-1886 fixed in global build stats plugin

### PR DESCRIPTION
**Pull request by MarkEWaite**
## SECURITY-1886 fixed in global build stats plugin

[288.vb_2c4a_0f138b_b_](https://plugins.jenkins.io/global-build-stats/releases/#version_288.vb_2c4a_0f138b_b_) is the release that fixes [SECURITY-1886](https://www.jenkins.io/security/advisory/2022-03-15/#SECURITY-1886)

### CODE REVIEW

File: resources/warnings.json [hunk 1]
```diff
@@ -11277,7 +11277,7 @@
     "versions": [
       {
         "lastVersion": "1.5",
-        "pattern": ".*"
+        "pattern": "1[.][0-5](|[.-].*)"
```
Comment (daniel-beck) [inline resources/warnings.json:11280]: Release 244 does not have the fix but would be considered unaffected by this pattern. Please also update the `lastVersion`.

**Review by daniel-beck — CHANGES_REQUESTED** [review]

File: resources/warnings.json (hunk 1 above)
Comment (MarkEWaite) [reply resources/warnings.json]: Fixed in 546a872514aeb2a76b4f2fac78244d63cf11c949 .  Thanks for detecting that mistake!

Comment (MarkEWaite) [reply resources/warnings.json]: @daniel-beck I believe this is ready for another review.  Thanks again for detecting my mistake in the `lastVersion` and the matching pattern.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
https://github.com/jenkinsci/global-build-stats-plugin/pull/38 is incomplete and amended by https://github.com/jenkinsci/global-build-stats-plugin/pull/52; while that's unmerged this issue is unresolved.

**Comment by Kevin-CB** [comment]
Time to reopen this PR, https://github.com/jenkinsci/global-build-stats-plugin/pull/52 got merged and a new release was released `288.vb_2c4a_0f138b_b_`

I didn't test the PR, but it was approved by another security team member.

**Comment by balakine** [comment]
@MarkEWaite a gentle reminder the fix is released and this issue can be reopened.

**Comment by MarkEWaite** [comment]
Thanks for the reminder @balakine and thanks very much for fixing the issue!

File: resources/warnings.json [hunk 2]
```diff
@@ -11276,8 +11276,8 @@
     "url": "https://www.jenkins.io/security/advisory/2022-03-15/#SECURITY-1886",
     "versions": [
       {
-        "lastVersion": "1.5",
-        "pattern": ".*"
+        "lastVersion": "282.v79ca_e079d1b_1",
+        "pattern": "1[.][0-5]|244[.]v27c8a_2e50a_34|269[.]v214f74360b_3a_|282[.]v79ca_e079d1b_1(|[.-].*)"
```
Comment (daniel-beck) [inline resources/warnings.json:11280]:
```suggestion
        "pattern": "(1|244|269|282)(|[.-].+)"
```

Lack of grouping the version numbers negates the purpose of the `(|[.-].+)` suffix.

Comment (MarkEWaite) [reply resources/warnings.json]: Thanks very much!  I have much to learn about regex pattern matching.  Fixed in 3bf7b744dd82c35c4e3197c40a2eb7d46cffd2fb

**Pull request merged**